dask/array/linalg.py

```diff
@@ -328,3 +328,89 @@ def svd(a, name=None):
     dask.array.linalg.tsqr: Actual implementation with citation
     """
     return tsqr(a, name, compute_svd=True)
+
+
+def _lu_diag(arr, pl, pu):
+    import scipy.linalg
+    res = scipy.linalg.lu(arr - np.dot(pl, pu))
+    return res
+
+def _lu_horizontal(arr, l):
+    return np.dot(np.linalg.inv(l), arr)
```

Comment:
Is `np.linalg.inv` efficient in the triangular case?  I did a quick benchmark and I think that `scipy.linalg.solve_triangular` might be faster here.